src/Common/Core/Impl/IO/FileSystem.cs

```diff
@@ -31,6 +31,9 @@ public sealed class FileSystem : IFileSystem {
         
         public void FileWriteAllBytes(string path, byte[] bytes) => File.WriteAllBytes(path, bytes);
 
+        public Stream CreateFile(string path) => File.Create(path);
+        public Stream FileOpenRead(string path) => File.OpenRead(path);
```

Comment:
To avoid having tons of overloads here for `OpenWrite` etc, let's wrap `File.Open` instead - it allows to specify the mode via enum.